stage/dotnet/dotnet-sdk/cli/Authentication/PersonalAccessToken.cs

```diff
@@ -18,7 +18,10 @@ public static async Task Run()
     {
         // Personal Access Token authentication
         var tokenProvider = new TokenProvider(Environment.GetEnvironmentVariable("GITHUB_TOKEN") ?? "");
-        var adapter = RequestAdapter.Create(new TokenAuthProvider(tokenProvider), "https://api.github.com");
+        var adapter = new RequestAdapter(new TokenAuthProvider(tokenProvider))
```

Comment:
The TokenAuthProvider would also be a good candidate to move to this pattern. I do like the design of building the adapter.

Reply:
Ohh let's discuss that.  I'd like to see what your thoughts are around doing this in the token provider - right now I am simply inheriting from the kiota one  to make our SDK surface more consistent and discoverable. There wasn't much else going on in that wrapper.

I did want to look at this pattern for the `ClientFactory`